assets/app/scripts/controllers/pod.js

```diff
@@ -79,10 +81,35 @@ angular.module('openshiftConsole')
       watches.push(DataService.watch("builds", $scope, function(builds) {
         $scope.builds = builds.by("metadata.name");
         Logger.log("builds (subscribe)", $scope.builds);
-      }));      
+      }));
+
+      (function initLogs() {
+        angular.extend($scope, {
+          logs: [],
+          logsLoading: true,
+          canShowDownload: false,
+          initLogs: initLogs
+        });
+        DataService.watchLogStream("pods/log", $routeParams.pod, $scope, function(line) {
+          $scope.$apply(function() {
+            $scope.logs.push({text: line});
+            $scope.canShowDownload = true;
+          });
+        }, function() {
+          $scope.$apply(function() {
+            $scope.logsLoading = false;
+          });
+        })
```

Comment:
But... I don't like `DataService.stream()` first, because it won't actually stream until calling `.go()`.  So perhaps the last method should be `.stream()` and the first should be... ??

This really should be a separate streamer service, not part of DataService.
```javascript
// misleading...
var streamer = DataService.stream('pods/log',$routeParams.pod, context);
/// better?
var streamer = Streamer.create('pods/log',$routeParams.pod, context);
streamer.onMessage(function(msg) { /* do stuff */  });
streamer.stream();            
```
But since it needs to share methods only available in DataService, it will have to live here for now.  Just need to think of a better name.  I guess `DataService.createStream()` then `streamer.stream()` may work...